src/Factory.php

```diff
@@ -5,7 +5,7 @@
 use Faker;
 
 /**
- * @template TObject as object
```

Comment:
"as" is basically a substitution. "of object" means that we cannot use non-objects